plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityCharged.java

```diff
@@ -1,135 +1,74 @@
 package com.denizenscript.denizen.objects.properties.entity;
 
 import com.denizenscript.denizen.objects.EntityTag;
-import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.Mechanism;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
-import org.bukkit.entity.Entity;
-import org.bukkit.entity.Guardian;
-import org.bukkit.entity.Vex;
-import org.bukkit.entity.WitherSkull;
-
-public class EntityCharged implements Property {
-
-    public static boolean describes(ObjectTag object) {
-        if (!(object instanceof EntityTag)) {
-            return false;
-        }
-        Entity entity = ((EntityTag) object).getBukkitEntity();
-        return entity instanceof WitherSkull
-                || entity instanceof Vex
-                || entity instanceof Guardian;
+import org.bukkit.entity.*;
+
+public class EntityCharged extends EntityProperty<ElementTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name charged
+    // @input ElementTag(Boolean)
+    // @description
+    // Controls whether an entity is charged:

```

Comment:
I again suggest you just copy from the original mech here, this variant is weird too